src/overload_hessian.jl

```diff
@@ -1,128 +1,99 @@
-## 1-to-1
-for fn in ops_1_to_1_s
-    @eval Base.$fn(t::HessianTracer) = promote_order(t)
-end
-for fn in ops_1_to_1_f
-    @eval Base.$fn(t::HessianTracer) = t
-end
-
-for fn in union(ops_1_to_1_z, ops_1_to_1_const)
-    @eval Base.$fn(::T) where {T<:HessianTracer} = empty(T)
-end
-
-## 2-to-1
-# Including second-order only
-for fn in ops_2_to_1_ssc
-    @eval function Base.$fn(a::HessianTracer, b::HessianTracer)
-        a = promote_order(a)
-        b = promote_order(b)
-        return distributive_merge(a, b)
+# TODO REVIEW: this keeps popping up so often that think it's worth an abstraction.  
+# It was previously called "promote_order".
+# The overly descriptive name is TEMPORARY.
+function tracer_from_hessian_OR_outer_product_OR_of_gradient(
+    t::T
+) where {T<:GlobalHessianTracer}
+    hessian_out = deepcopy(t.hessian)
+    hessian_out = outer_product_or!(hessian_out, t.gradient, t.gradient)
+    return T(t.gradient, hessian_out)
+end
```

Comment:
For me this should just be

```julia
elementwise_or(t.hessian, outer_product_or(t.gradient, t.gradient)) 
```

No need for in-place stuff

Reply:
And thus it doesn't need a specific name

Reply:
Will do, see [here](https://github.com/adrhill/SparseConnectivityTracer.jl/pull/59#discussion_r1597668250).